app/microservice_tables.py

```diff
@@ -0,0 +1,148 @@
+from structlog import get_logger
+
+logger = get_logger('fsdr-ui')
+
+class Field:
+    def __init__(self,
+            database_name,
+            search_type="input_box",
+            search_options=None,
+            column_name=None,
+            accordion=False,
+            dropdown_options=None):
+
+        self.database_name =  database_name
+        self.column_name = self.create_column_name(column_name)
+        self.search_type = search_type
+        self.dropdown_options = self.format_dropdown_options(dropdown_options) 
+        self.accordion = accordion
+        self.previous_value = "" 
+
+    def create_column_name(self,column_name):
+        if column_name == None:
+            column_name = self.database_name.replace("_"," ").title()
+        else:
+            column_name = column_name
+        return(column_name)
+
+    def format_dropdown_options(self,dropdown_options,selected_value='blank'):
+        if dropdown_options != None:
+            final_dropdowns = [{'value':'blank', 'text':'Select a status', "disabled": True, }]
+            for option in dropdown_options:
+                entry = {'value': option, 'text': option}
+                final_dropdowns.append(entry)
+            
+            for each_dict in final_dropdowns:
+                if each_dict['value'] == selected_value:
+                    each_dict['selected'] = True
+ 
+            return final_dropdowns 
+
+    def refresh_selected_dropdown(self, selected_value):
+        dropdown_values = [i.get('value') for i in self.dropdown_options ]
+        self.dropdown_options = self.format_dropdown_options(
+                dropdown_values,
+                selected_value=selected_value)
+
+
+def load_cookie_into_fields(Fields,previous_criteria):
+    for field in Fields:
+        if field.database_name in previous_criteria.keys():
+            if  field.search_type == "input_box":
+                field.previous_value = previous_criteria.get(field.database_name)
+            elif field.search_type == "dropdown":
+                field.refresh_selected_dropdown(previous_criteria.get(field.database_name))
+    return Fields
+
+def get_fields(service_name):
+    # Set the default parameters for some services
+    status_options =  [ 
+            "CREATE",
+            "SETUP",
+            "UPDATE",
+            "LEAVER",
+            "LEFT",
+            "COMPLETE",]
+
+    fields =  []
```

Comment:
Instead of storing it in a variable (which doesn't need to be pre-initialized), you could simply return directly from inside the if-statement.

Reply:
Adjusted